core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -59,13 +60,14 @@ private BootstrapCheck() {
      * checks
      *
      * @param settings              the current node settings
+     * @param dataDirectoryNumber   the integer number of the directories who's locks under the data directory we were able to take
      * @param boundTransportAddress the node network bindings
      */
-    static void check(final Settings settings, final BoundTransportAddress boundTransportAddress) {
+    static void check(final Settings settings, final int dataDirectoryNumber, final BoundTransportAddress boundTransportAddress) {
```

Comment:
It didn't feel great to me either. I'd originally thought of passing the `NodeEnvironment` in here and thinking of is as a pool of stuff to check. But that doesn't work out well because `NodeEnvironment` is so large. It is too much too pass in. Not to mention that you can't mock it effectively.

I can absolutely pass in an interface that lets us pull the data we need and then implement that interface at the call site, but I'm not sure that it is much better. So I went with the simplest, if least forward looking option.